app/models/slack_account.rb

```diff
@@ -13,8 +13,13 @@ def self.attributes_from_auth_hash(auth_hash)
   end
 
   def self.create_from_auth_hash(auth_hash)
+    # Extract the domain from the url
+    domain = URI.parse(auth_hash[:extra][:raw_info][:url]).host.gsub(/\.slack.com/, "")
+
+    # TODO: This should just be an upsert
     team = SlackTeam.find_or_initialize_by(id: auth_hash[:info][:team_id]) do |t|
-      t.domain = auth_hash[:info][:team_domain]
+      # team_domain doesn't appear to be set, so we fall back to team_id
```

Comment:
Not sure I get this comment

Reply:
Whoops, that was left in from my original exploration.